### PR TITLE
tech: Create ToggleCard UI component

### PR DESCRIPTION
**Pull request by Youakeem**
## Describe your changes
Create a new version of the `ToggleCard` component in the `ui` package with vanilla-extract and composable API.

## Justify why they are needed
Building block for the `ProductCard` components

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by Youakeem** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4733?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4734** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4734?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4733** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4733?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4732** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4732?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4730** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4730?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4729** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4729?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4728** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4728?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4700** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4700?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4699** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4699?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4698** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4698?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4697** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4697?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4696** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4696?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4710** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4710?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4723** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4723?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4709** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4709?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4708** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4708?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4707** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4707?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4706** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4706?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4705** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4705?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4704** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4704?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4703** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4703?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Youakeem and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by Youakeem** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4733?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4734** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4734?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4733** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4733?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4732** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4732?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4730** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4730?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4729** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4729?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4728** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4728?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4700** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4700?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4699** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4699?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4698** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4698?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4697** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4697?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4696** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4696?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4710** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4710?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4723** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4723?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4709** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4709?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4708** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4708?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4707** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4707?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4706** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4706?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4705** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4705?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4704** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4704?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4703** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4703?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Youakeem and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/ui/src/components/ToggleCard/ToggleCard.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import { type ReactNode, type ComponentProps } from 'react'
+import { xStack } from '../../patterns'
+import { Card } from '../Card/Card'
+import { Switch } from '../Switch/Switch'
+import { Text } from '../Text/Text'
+import { toggleCardSwitchStyles } from './ToggleCard.css'
+
+type RootProps = ComponentProps<typeof Card.Root>
+
+const ToggleRoot = ({ children, ...props }: RootProps) => {
```
Comment (guilhermespopolin) [inline packages/ui/src/components/ToggleCard/ToggleCard.tsx:11]: ocd 😬 
```suggestion
type RootProps = ComponentProps<typeof Card.Root>
const ToggleRoot = ({ children, ...props }: RootProps) => {
```

File: packages/ui/src/components/ToggleCard/ToggleCard.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+const ToggleSwitch = (props: ToggleSwitchProps) => {
+  return (
+    <Card.Aside>
+      <Switch className={toggleCardSwitchStyles} {...props} />
+    </Card.Aside>
+  )
+}
+
+type ToggleLabelProps = ComponentProps<'label'> & {
+  pre?: ReactNode
```
Comment (guilhermespopolin) [inline packages/ui/src/components/ToggleCard/ToggleCard.tsx:29]: Is this really necessary? I guess there's no usage for it in the stories right?!

**Review by guilhermespopolin — APPROVED** [review]

File: packages/ui/src/components/ToggleCard/ToggleCard.tsx (hunk 1 above)
Comment (Youakeem) [reply packages/ui/src/components/ToggleCard/ToggleCard.tsx]: I got ya!

**Comment by Youakeem** [comment]
### Merge activity

* **Sep 19, 9:20 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4733) rebased this pull request as part of a merge.

**Pull request merged**